service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java

```
@@ -166,8 +171,8 @@ public void batchUpsert(UUID instanceId, RecordType recordType, List<Record> rec
 		} catch (DataAccessException e) {
 			if (e.getRootCause()instanceof SQLException sqlEx) {
 				checkForMissingRecord(sqlEx);
-				throw e;
 			}
+			throw e;
```

I noticed while testing that we were swallowing exceptions here.